viewing_party/party.py

```diff
@@ -1,23 +1,212 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(title: str, genre: str, rating: float) -> dict: 
+    movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating,
+    }
+    
+    if title == None or genre == None or rating == None:
+        return None
+        
+    return movie
+
+
+def add_to_watched(user_data: dict, movie: dict) -> dict:    
+    user_data["watched"].append(movie)
+
+    return user_data 
+
+
+def add_to_watchlist(user_data: dict, movie: dict) -> dict:  
+    user_data["watchlist"].append(movie)
 
+    return user_data
+
+
+def watch_movie(user_data: dict, title: str) -> dict:
+    user_watchlist = user_data["watchlist"]
+    
+    for movie in range(len(user_watchlist)):
+        if user_watchlist[movie]["title"] == title:
+            user_data["watched"].append(user_watchlist[movie])
+            del user_watchlist[movie]
+
+    return user_data 
+
+   
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data: dict) -> float:
+    user_ratings_list = []
+    
+    for movie in user_data["watched"]:
+        user_ratings_list.append(movie["rating"])
+
+    try:
+        average_rating = sum(user_ratings_list) / len(user_data["watched"])
+        return average_rating
+    except(ZeroDivisionError):
+        return 0.0
+
+
+def get_most_watched_genre(user_data: dict) -> str:
+    if len(user_data["watched"]) == 0:
+        return None

```

Comment:
In addition to checking if `user_data["watched"]` is an empty list, the `not` operator will check if it is any variation of a _falsey_ value.

```python
if not user_data["watched"]:
    return None
```